### PR TITLE
Wide range of awards covering many different aspects of gameplay

### PR DESCRIPTION
**Pull request by Wuzzy2**
This PR adds quite a lot of new awards.

The general idea I have with this mod is that there are eventually many different awards which are realistic to get, not a few awards which take absurd amounts of time because of crazy high numbers. I tried to cover many different aspects of gameplay, moving the attention to different areas, and avoiding boring and repetitive tasks, in hope to keep the player busy. Award numbers should be capped a numbers where it starts too boring.
Digging awards should be stopped at a point where it becomes either frustratingly boring or digging starts to eat away your diamond tools because of the sheer number. Both would be frustrating and should be avoided.

These are the ideas I have for this mod, and this PR adds and change awar

Playing through all awards except the nyan cat one might take one realtime day or two, but I am hesitant to increase numbers just to make it last longer. That's really just Minetest Game running out of gameplay depth; there isn't much to do about in the awards mod.

Things covered in awards:
* Most awards are crafting and mining
* Fire (create and destroy)
* Farming (much higher numbers)
* Lumberjack (was nerfed)
* Another “discovery” award, this time for snow lands (not just glacier)
* Filthy Rich demands crafting of 24 gold block stairs (=216 gold lumps or so). Not sure if this is too much; if yes might reduce by 6. But I though this award is fun because it gives a reason to craft gold which is otherwise rather useless in vanilla Minetest Game, and to show off pure luxury xD. stairs were chosen because craft is not reversible. It can also be seen as a successor to Gold Rush
* Crafting dyes (actually quite interesting, as you need to explore different areas to get the flowers. It's a bummer that same color → same color crafts still count, therefore subverting the achievement)
* Crafting new blocks

Things I would maybe change in this PR:
* Farming awards only cover wheat, but not cotton, but I feel adding cotton awards is just redundant; the player will probably plant both sooner or later.

I have playtested most of my awards in a completely new world to check for fun/boredom levels.
I think this works out quite well for me, I actually managed re-discover Minetest Game after a long time of mostly ignoring it.

I really liked doing the new farmer awards, going from a single wheat plant to a huge farm is pretty fun and doesn't get boring **too** soon. As your farm will grow faster and fast, the high numbers are justified and realistic. I even considered to add an award for digging 3125 wheat (after 625), a number which I would reach in my next harvest, but it took some time of course. At that point, the farm was already starting to eat away my diamond swords, my last harvest cost me almost a new diamond sword.

Also an award for Unified Inventory is added (4 large bags). I know, I know, it would be better to put it into that mod instead, but this is not possible because this would make a circular dependency.

I have also tried to balance out some existing awards.
I have massively nerfed the Lumberjack award; esp. because the numbers were too extreme. See the full commit description to read why.

I also changed the icons of some awards to be more consistent. There is a new icon overlay for “level-style” achievements (same award, bigger numbers). Tell me if you like it.

Finally I also attempted to add an award “Risk your Life” which is like Firefighter (put out fires), but you have to stand very close to the fire you put out. I failed to implement this award, so it is not included.

With this PR, there is now a total of ca. 68 awards.
And I already have ideas for many more awards which I haven't added because they demand more award types.

Don't worry about the many dependencies, they are just listed because this mod failed to list them beforehand. Getting dependencies right for real will probably fit better in another commit.

Hopefully the commit list is not too chaotic for you. ;-)
This PR puts many different things together, sorry for that. I try to be more strict on seperation next time. :-/

Oh, all new textures are WTFPL, of course.

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -93,18 +103,78 @@ if minetest.get_modpath("moreblocks") then
...
+		}
+	})
+end
+
+if minetest.get_modpath("fire") ~= nil then
+	awards.register_achievement("awards_pyro", {
+		title = S("Pyromaniac"),
+		description = S("Craft 8 times flint and steel."),
+		icon = "fire_flint_steel.png",
+			trigger = {
```
Comment (rubenwardy) [inline init.lua]: Bad indentation

File: init.lua [hunk 2]
```diff
@@ -93,18 +103,78 @@ if minetest.get_modpath("moreblocks") then
...
+		icon = "awards_ui_bags.png",
+		trigger = {
+			type = "craft",
+			item = "unified_inventory:bag_large",
+			target = 4
+		}
+	})
+end
+
+if minetest.get_modpath("fire") ~= nil then
```
Comment (rubenwardy) [inline init.lua]: ~= nil is redundant

File: init.lua [hunk 3]
```diff
@@ -93,18 +103,78 @@ if minetest.get_modpath("moreblocks") then
...
+			item = "fire:flint_and_steel",
+			target = 8
+		}
+	})
+	if minetest.setting_getbool("disable_fire") ~= true then
+		awards.register_achievement("awards_firefighter", {
+			title = S("Firefighter"),
+			description = S("Put out 1000 fires."),
+			icon = "awards_firefighter.png",
+				trigger = {
```
Comment (rubenwardy) [inline init.lua]: Here too

File: init.lua [hunk 4]
```diff
@@ -331,8 +415,8 @@ awards.register_achievement("award_mine3", {
 awards.register_achievement("award_mine4", {
 	title = S("Master Miner"),
 	description = S("Dig 10,000 stone blocks."),
-	icon = "awards_miniminer.png",
-	background = "awards_bg_mining.png",
+	icon = "awards_miniminer.png^awards_level3.png",
+	background = "awards_bg_mining.png.png",
```
Comment (rubenwardy) [inline init.lua]: Is this a mistake?

**Comment by rubenwardy** [comment]
Mostly good. The big number of dug blocks aren't meant to be things you aim for, but would happen after a while/long time of playing

As for unified inventory - the awards mod should use the observer pattern as much as possible. The unified inventory code shouldn't care about awards, imo. Same with other mods, for example a call to awards is a bit weird to find in a mobs update function

**Comment by rubenwardy** [comment]
I'm not sure what the aim of this mod is. I like the idea of making it reward players for exploring new things and also ending up in weird contexts. I also like the idea of it rewarding players for great feats, but apart from dig/place/etc that's difficult. Shame it's almost impossible to make "build a castle/prison/house" reward

**Comment by Wuzzy2** [comment]
I still think 100000 tree blocks is VERY unrealistic, read my commit description. Also, it's limited to normal trees only, other tree blocks don't count. Also, at some point you have so much wood there is no really need to dig new trees for a long long time, so it is also unlikely for the award to “happen”. Chopping wood is also a lot less fun than mining; chopping wood is rather slow compared to mining and there is nothing interesting in it.
 Thus the nerfing. If you want to have an award for long play, it should be something a bit more interesting and it should be something that involves less than 999 diamonds worth of tools. xD
Note I am OK with 10000 stone blocks award, this number seems fair enough since you will often dig stone and is of course much more common than tree and you can always try to dig for mineral, so it is not that repetetive as woodchopping.
Yes, only having dig/place/craft is not much, so support for other award types will help a lot. I already have ideas, but lets not get too much off-topic here.

And oh shit, I should really check my code before submitting. Of course `.png.png` is not intentional. ]:-<
Maybe this PR was a little bit too rushed.

**Comment by rubenwardy** [comment]
Yeah, I agree. That's rather too much

File: init.lua [hunk 5]
```diff
@@ -506,3 +650,158 @@ awards.register_achievement("awards_mossycobble", {
...
+if minetest.get_modpath("vessels") ~= nil then
+	awards.register_achievement("award_vessels_shelf", {
+		title = S("Glasser"),
+		icon = "vessels_shelf.png",
+		description = S("Craft 14 vessels shelves."),
+		trigger = {
+			type = "craft",
+			item= "vessels:shelf",
+			target = 14
+		}})
```
Comment (rubenwardy) [inline init.lua]: Inconsistent style

**Comment by Wuzzy2** [comment]
OK, I fixed the double `.png` thing and the other code style issues you had.

**Comment by sofar** [comment]
I like where this is going, any reason not to merge it?

File: init.lua [hunk 6]
```diff
@@ -506,3 +655,168 @@ awards.register_achievement("awards_mossycobble", {
...
+		description = S("Craft 14 vessels shelves."),
+		trigger = {
+			type = "craft",
+			item= "vessels:shelf",
+			target = 14
+		}
+	})
+end
+
+	awards.register_achievement("award_furnace", {
```
Comment (rubenwardy) [inline init.lua]: bad indentation here

File: init.lua [hunk 7]
```diff
@@ -506,3 +655,168 @@ awards.register_achievement("awards_mossycobble", {
...
+
+	awards.register_achievement("award_chest", {
+		title = S("Treasurer"),
+		description = S("Craft 15 chests."),
+		icon = "default_chest_front.png",
+		trigger = {
+			type = "craft",
+			item= "default:chest",
+			target = 15
+		}})
```
Comment (rubenwardy) [inline init.lua]: indentation here doesn't match rest of file

**Comment by rubenwardy** [comment]
will fix on merge

**Pull request merged**